rclpy/actions/minimal_action_server/examples_rclpy_minimal_action_server/server_single_goal.py

```diff
@@ -0,0 +1,123 @@
+# Copyright 2018 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import threading
+import time
+
+from example_interfaces.action import Fibonacci
+
+import rclpy
+from rclpy.action import ActionServer, CancelResponse, GoalResponse
+from rclpy.callback_groups import ReentrantCallbackGroup
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+
+
+class MinimalActionServer(Node):
+    """Minimal action server that processes one goal at a time."""
+
+    def __init__(self):
+        super().__init__('minimal_action_server')
+        self._goal_handle = None
+        self._goal_lock = threading.Lock()
+        self._action_server = ActionServer(
+            self,
+            Fibonacci,
+            'fibonacci',
+            execute_callback=self.execute_callback,
+            goal_callback=self.goal_callback,
+            cancel_callback=self.cancel_callback,
+            callback_group=ReentrantCallbackGroup())
+
+    def destroy(self):
+        self._action_server.destroy()
+        super().destroy_node()
+
+    def goal_callback(self, goal_request):
+        """Accepts or rejects a client request to begin an action."""
+        self.get_logger().info('Received goal request')
+        with self._goal_lock:
+            # This server only allows one goal at a time
+            if self._goal_handle is not None and self._goal_handle.is_active:
+                self.get_logger().info('Aborting previous goal')
+                # Abort the existing goal
+                self._goal_handle.set_aborted()
+
+        return GoalResponse.ACCEPT_AND_EXECUTE
+
+    def cancel_callback(self, goal):
+        """Accepts or rejects a client request to cancel an action."""
+        self.get_logger().info('Received cancel request')
+        return CancelResponse.ACCEPT
+
+    def execute_callback(self, goal_handle):
+        """Executes the goal."""
+        self.get_logger().info('Executing goal...')
+        with self._goal_lock:
+            self._goal_handle = goal_handle
```

Comment:
Is it possible for two goals to get through `goal_callback` before the first reaches `execute_callback`? In that case one of them will overwrite the other in `self._goal_handle`

Reply:
Fixed with the use of `handle_accepted_callback` b0c739a